### ISSUE TITLE
Mist password wrong error 0.7.6

### ISSUE THREAD

**Issue by teasider** [ethereum/mist#923]
This is the information ive managed to gather:
wallet was created with Mist, about a month ago. Funds were sent to it.

around the 22nd of June, I tried using myetherwallet to view info on the wallet and I encountered a wrong password error a few times.

After trying again and again, I installed the chrome extensions and it worked, I could view the wallet's info.

Yesterday I tried loading the same wallet ( I didnt re-save or change the password) into Mist - I can see the funds, but when trying to send - I get a "wrong password" error. 

This seems very odd as i was able to see the content of the wallet in MyEtherwallet on the 22nd of june (not anymore.)

Any suggestions on what might be causing this? I ruled out the fact that im using a mistaken password as creating a wallet in Mist forces you to re-type it when creating a new wallet.

**Comment by luclu** [comment]
@teasider could you please check if you have the same account several times in your keystore folder?
https://github.com/ethereum/mist/issues/780

**Comment by teasider** [comment]
@luclu Hey, the wallets were kept externally.
This is a new installation of Mist - after installing I ran it, closed it and added the wallet into the keystore directory. 
So right now I have two wallets in there, the one Mist created now, and the old backedup one with the "wrong password" error.

**Comment by ferni** [comment]
It happened to me in the new Mist beta: I created a new account for testing "Stake Voice" dapp, with a four letter password. I sent 0.001 eth from another account. Then I created a statement in Stake Voice and clicked to vote "Yes". When entering the password, I get a "Wrong password" message. Then I went back to the wallet and attempted to send 0.0001 eth back to my main account, again I get "wrong password". Then I attempted to get back the funds with the "send everything" checkbox, and it worked.

I also created another account, and tried to vote on Stake Voice with it (despite the fact the the account has 0 ether), when entering the password I also got the "wrong password" message.

I think the issue could be that the "wrong password" message is a default message, and the error was actually "Not enough ether" or "Not enough ether to pay for gas", but one mistakenly gets a "wrong password" message.

**Comment by luclu** [comment]
@teasider @ferni it seems to be related to this issue: https://github.com/ethereum/mist/issues/664
I will look into this today.

**Comment by teasider** [comment]
Is this relevant if I just try to unlock my wallet using geth, and still get a wrong passphrase error?

**Comment by luclu** [comment]
@teasider yes! Please check the follow recommendation https://github.com/ethereum/mist/issues/669. 
What error message do you encounter?
Are you able to unlock it with myethereumwallet.com?

**Comment by luclu** [comment]
Apparently current "wrong password" alert is broken, as the error string emitted by geth did change.
Possible errors are as following https://github.com/ethereum/mist/issues/664:
>
 - `could not decrypt key with given passphrase` (wrong password)
 - `multiple keys match address` (multiple copies of the same account)
 - `Insufficient funds for gas * price + value` (not enough ether in etherbase account to invoke contract transaction)
 - `Unable to connect to socket: timeout` (node terminated, etc; used to be `CONNECTION ERROR`)

The ongoing PR is https://github.com/ethereum/mist/pull/959.

In future releases we propose to use [error codes](https://github.com/ethereum/wiki/wiki/JSON-RPC-Error-Codes-Improvement-Proposal) to ensure consistency and compatibility to other nodes.

**Comment by teasider** [comment]
@luclu In geth I'm getting "wrong passphrase", in mist I get "wrong password" and I cant unlock it with myetherumwallet.com either. (here it just takes a long while until it responds and then the same error.

My question is - is it possible that while creating the wallet - some bug occured that inserted\removed a character?
I only ever use two passwords, and ive tried using them, and running ethcracker with a password variator that generates my passwords with\without letters and diff combinations.

**Comment by luclu** [comment]
@teasider what version of are you currently running? Make sure it's the latest.

**Comment by teasider** [comment]
@luclu 
Mist 0.8.0
Geth 1.4.9

**Comment by luclu** [comment]
>  Are you able to unlock it with myethereumwallet.com?

**Comment by luclu** [comment]
How did using myethereumwallet.com turn out?
Also please have a try with `0.8.1` as the error messages are now correct.

**Comment by teasider** [comment]
Nop, Both gives me a wrong password error. (myetherwallet and 0.8.1).
I dont remember if myetherwallet asks you to re-enter your password when creating a wallet - even thu I've tried EVERY possible combination of mistakes around my password to see if i may clicked another key or not - still nothing.

**Comment by DavidMc0** [comment]
I've had the same issue on Mist 0.8.1

I know the password is that I set, and it's unlikely that I made a typo on 2 occasions.  It won't open on Mist or Myetherwallet with the password I believe I set.

I've been trying all kinds of combinations of possible typos etc, and no luck yet - just started a 77 hour brute force after many smaller targeted attempts.

The only possibility I can think of is corruption during the creation process.  Every other account I've created on the same computer & installation works fine.  Of course it's possible that it's human error - brain farts do happen, and getting muddled up & typing the wrong thing twice without realising it isn't out of the realms of possibility.

Thought I'd post to say I'm experiencing the same, in case there is an issue other than human error.

**Comment by teasider** [comment]
@DavidMc0 you have concluded my problem exactly like it is. Same thing exactly.
I'm already after 80 hours of brute force using a password variator that takes in account mis-types by one letter or two letters.

I've also created multiple wallets on the same computer - all work great, with the same exact password.
Thanks for sharing, makes me feel less crazy on the subject.

**Comment by DavidMc0** [comment]
It does seem strange.

Hope you get some luck with the brute force.  Even if this were a bug & not human error, I'm feeling stupid for sending assets to the address without checking I can use it first!

**Comment by teasider** [comment]
I had no luck with the brute force, So I didnt have a mistake with 1 &/or 2 letters - which obviously means something got corrupted.

I was able to "View wallet" on myetherwallet ONCE, after it didnt work a few times - but I cant get it open again. (foolish of me not to move funds back then)

**Comment by testerguy** [comment]
Hi, I'm getting the same problem with both Mist and Geth. I am sure I noted the passwords down correctly.

I unfortunately did not test to see if the passwords worked before sending a large amount of ether to them, because I was so confident that I had noted them down correctly and that the software would work.

**Comment by DavidMc0** [comment]
I had no luck with days worth of brute forcing using possible typos.  No idea what happened given the password had to be entered twice.

The longer I try, the more I think this is likely to be a bug, not human error.

@testerguy sorry to hear about your issue - lets hope a fix can be found.

**Comment by hiddentao** [comment]
So sorry to hear this happening for you guys. My understanding was that as of #182 the carriage-return bug preventing correct password verification in Mist had been fixed. I can have a look at the creation process (atleast from Mist's side) to ensure everything is as it should be. 

Just from my own experience, I'm creating accounts and having to send ether from them all the time. And as far as I can tell password creation works ok. But will  take a closer look at the code now.

**Comment by luclu** [comment]
Could you please give the `geth account update` command a try?
This might be helpful as [this post](https://github.com/ethereum/mist/issues/982#issuecomment-247409749) suggests.

**Comment by testerguy** [comment]
@luclu thanks I will try it and see how it goes

**Comment by teasider** [comment]
@DavidMc0 Did you manage to release your funds from the locked account?
still no luck on my side.

**Comment by DavidMc0** [comment]
@teasider I have managed to release my funds, but unfortunately I wasn't paying attention when I randomly got the right combination & the transaction progressed.

I had spent ages trying brute forcing, and after basically giving up hope I was just manually trying a whole load of combinations at random.  I did eventually get a combination that worked.  Previously, I had been paying attention so that if I guessed the correct combination, I could see what the error was and what it related to, but I was unable to replicate the 'correct' passphrase once I typed it once (which thankfully ran the transaction to send the funds).

I tried again for a while to figure out what the passphrase was, in case the error related to a specific special character (which is my suspicion), but unfortunately I couldn't replicate the correct phrase.

So, yes I have released my funds, but apologies that I can't point to what the error was, except that I'm confident that the phrase that unlocked the wallet is not the phrase I would have chosen / typed as my password.

Hope you manage to get some luck like I did.

**Comment by teasider** [comment]
@DavidMc0 Glad to hear you managed to release them! 
The combinations you used, were those made up of your previous passwords or just random words? That'll be like the luck needed to win the lottery :)

**Comment by DavidMc0** [comment]
@teasider It was based around the password I thought it would be, but with typos / errors, & substituted special characters, so not just random words, which would indeed be highly improbable!

**Comment by Sonatix** [comment]
@DavidMc0  Can you (or someone) share geth or mist cracker software to brutforce my wallet? Have exactly same problem (always used only 2 passwords and both are wrong for now :/ )

**Comment by DavidMc0** [comment]
@Sonatix the software I used is called 'password variator' to create variations of the password I thought I had used, then used 'ethcracker' to run those password variations against my wallet file. Took a while to get to know how to use these tools, but worth the effort if it helps recover funds.

Hope you have more luck with it that I did, though that may be due to the fact I have a UK keyboard layout, and password variator doesn't allow for this (assumes US layout), so some of the typo suggestions would have been highly unlikely.

Just to be clear, you need to know what password you thought you used to have any chance of success with this, as it just varies your password with typos etc.  If you have no idea what your password is, unfortunately you have a near zero chance of guessing it.

**Comment by Sonatix** [comment]
Thank you for the answer :) I will try it.
